flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/AdaptedRestartPipelinedRegionStrategyNG.java

```diff
@@ -148,19 +145,10 @@ private RestartCallback createResetAndRescheduleTasksCallback(final long globalM
 			} catch (GlobalModVersionMismatch e) {
 				throw new IllegalStateException(
 					"Bug: ExecutionGraph was concurrently modified outside of main thread", e);
-			} catch (Exception e) {
-				throw new CompletionException(e);
-			}
-		};
-	}
-
-	private BiFunction<Object, Throwable, Object> failGlobalOnError() {
-		return (Object ignored, Throwable t) -> {
-			if (t != null) {
+			} catch (Throwable t) {
 				LOG.info("Unexpected error happens in region failover. Fail globally.", t);
 				failGlobal(t);
```

Comment:
I think it is better to let the exception handling happen where it has been. Usually it is a good idea to do the exception handling at the highest possible level.